go-controller/pkg/clustermanager/egressip_controller_test.go

```diff
@@ -1579,91 +1579,6 @@ var _ = ginkgo.Describe("OVN cluster-manager EgressIP Operations", func() {
 			err := app.Run([]string{app.Name})
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 		})
-
-		ginkgo.It("egress node update should not mark the node as reachable if there was no label/readiness change", func() {
```

Comment:
So the original code that bypasses adding/updating a node if the ready states are equal: https://github.com/ovn-org/ovn-kubernetes/pull/1784/commits/1f101eb655c4040fc7225cf4b28f544e7df32533#diff-796adb9d8a9b5b5636dc81e7350a00ef143e6e5bcb52ec30dc350c866cd9208cR784

This pre-dates the reachability check. So this test case was added later. This test case seems flawed to me in the first place. We should use any event to trigger us to check and ensure that the node is added, whether or not that is a node update or a reachability check success.

Reply:
yes totally agree